src/main/java/com/google/sps/user/User.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.user;
+
+import org.jetbrains.annotations.Nullable;
+
+/** Stores the data related to one user. */
+public class User {
+    private final String id, email;
+    private final boolean publicPortfolio;
+    @Nullable 
+    private final String name, selfIntroduction, imgUrl;
+
+    public static class Builder {
+        // Required.
+        private final String id, email;
+        // Optional.
+        private String name, selfIntroduction, imgUrl;
```

Comment:
Maybe use Optional<>

AutoValue can handle this for you

Reply:
Added `Optional<>`

Reply:
Removed optional<>, because it's not supported by Datastore.